contracts/RelayerHub.sol

```diff
@@ -8,88 +8,189 @@ import "./System.sol";
+        require(managersRegistered[msg.sender], "manager not registered");
+        _;
+    }
+
+    modifier noProxy() {
+        require(msg.sender == tx.origin, "no proxy is allowed");
+        _;
+    }
+
+    event relayerRegister(address _relayer);
+    event relayerUnRegister(address _relayer);
+    event paramChange(string key, bytes value);
+
+    event removeManagerByGovEvent(address _removedManager);
+    event addManagerByGovEvent(address _addedManager);
+    event registerManagerEvent(address _registeredManager);
+    event addRelayerEvent(address _relayerToBeAdded);
+    event removeRelayerEvent(address _removedRelayer);
+
+
+    function init() external onlyNotInit {
+        requiredDeposit = INIT_REQUIRED_DEPOSIT;
+        dues = INIT_DUES;
+        addInitRelayer(WHITELIST_1);
+        addInitRelayer(WHITELIST_2);
+        alreadyInit = true;
+    }
+
+    function addInitRelayer(address addr) internal {
+        managers[addr] = manager(requiredDeposit, dues);
+        managersRegistered[addr] = true;
+        relayManagersExistMap[addr] = true;
+        managersAndRelayers[addr] = addr; // fixme current relayer
+        relayerExistsMap[addr] = true;
+    }
+
+    /*********************** Param update ********************************/
+    function updateParam(string calldata key, bytes calldata value) external override onlyInit onlyGov {
+        if (Memory.compareStrings(key, "requiredDeposit")) {
+            require(value.length == 32, "length of requiredDeposit mismatch");
+            uint256 newRequiredDeposit = BytesToTypes.bytesToUint256(32, value);
+            require(newRequiredDeposit > 1 && newRequiredDeposit <= 1e21 && newRequiredDeposit > dues, "the requiredDeposit out of range");
+            requiredDeposit = newRequiredDeposit;
+        } else if (Memory.compareStrings(key, "dues")) {
+            require(value.length == 32, "length of dues mismatch");
+            uint256 newDues = BytesToTypes.bytesToUint256(32, value);
+            require(newDues > 0 && newDues < requiredDeposit, "the dues out of range");
+            dues = newDues;
+        } else if (Memory.compareStrings(key, "addManager")) {
+
+            require(value.length == 20, "length of manager address mismatch");
+            address newManager = BytesToTypes.bytesToAddress(20, value);
+            addManagerByGov(newManager);
+
+        } else if (Memory.compareStrings(key, "removeManager")) {
+
+            require(value.length == 20, "length of manager address mismatch");
+            address payable managerAddress = payable(BytesToTypes.bytesToAddress(20, value));
+            removeManagerByGov(managerAddress);
+
+        } else {
+            require(false, "unknown param");
+        }
+        emit paramChange(key, value);
+    }
+
+    function removeManagerByGov(address payable managerToBeRemoved) internal {
+        removeManagerHelper(managerToBeRemoved);
+    }
+
+    function removeManager() external onlyRegisteredManager {
```

Comment:
Is this `onlyRegisteredManager` modifier necessary, since `removeManagerByGov` dont have this, and `removeManagerHelper` will also do duplicate check.